### PR TITLE
Add better task setup

### PR DESCRIPTION
**Pull request by kirkone**
- Add overloads to `AddScheduledTask` to provide options
- Add extension method `AddScheduledTaskOptions`
- Add auto register for `SchedulerHostedService`

+version: feature

### CODE REVIEW

**Comment by kirkone** [comment]
The readme is not updated yet. This had to be done before merging!

File: src/KK.DotNet.BackgroundTasks.Scheduled.Web.Sample/Startup.cs [hunk 1]
```diff
@@ -37,32 +37,58 @@ public void ConfigureServices(IServiceCollection services)
...
             services.AddScheduledTask<SampleTask>();
-            services.AddScheduledTask<SuperSampleTask>();
 
-            // Setup the Host for the scheduled tasks
-            // without this the tasks will never run.
-            services.AddHostedService<SchedulerHostedService>();
+            // You can also configure the task when adding it
+            services.AddScheduledTask<SuperSampleTask>(options =>
+                {
+                    options.Schedule = "*/10 * * * * *";
```
Comment (paule96) [inline src/KK.DotNet.BackgroundTasks.Scheduled.Web.Sample/Startup.cs:88]: Add a comment that describe the `"*/10 * * * * *"`

File: src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs [hunk 2]
```diff
@@ -1,12 +1,99 @@
...
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return AddService<TScheduledTask>(services);
+        }
+
+        private static IServiceCollection AddService<TScheduledTask>(
+            IServiceCollection services
+        )
+            where TScheduledTask : class, IScheduledTask
+        {
+            services.TryAddSingleton<IHostedService, SchedulerHostedService>();
```
Comment (paule96) [inline src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs]: That's wrong. You need your own Interface

File: src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs [hunk 3]
```diff
@@ -1,12 +1,99 @@
...
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            if (configuration == null) throw new ArgumentNullException(nameof(configuration));
+
+            var options = new ScheduledTaskOptions<TScheduledTask>();
+            configuration.Bind(options);
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return options;
+        }
+
+        public static IScheduledTaskOptions<TScheduledTask> AddScheduledTaskOptions<TScheduledTask>(
```
Comment (paule96) [inline src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs:26]: All overloads of this method use the same logic. Please merge it.

File: src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs [hunk 4]
```diff
@@ -1,12 +1,99 @@
...
+            if (services == null) throw new ArgumentNullException(nameof(services));
+            if (configureOptions == null) throw new ArgumentNullException(nameof(configureOptions));
+
+            var options = new ScheduledTaskOptions<TScheduledTask>();
+            configureOptions.Invoke(options);
+            services.AddSingleton<IScheduledTaskOptions<TScheduledTask>>(options);
+            return options;
+        }
+
+        public static IServiceCollection AddScheduledTask<TScheduledTask>(
```
Comment (paule96) [inline src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs:54]: All overloads of this method use the same logic. Please merge it.

**Review by paule96 — CHANGES_REQUESTED** [review]
please fix the comments

File: src/KK.DotNet.BackgroundTasks.Scheduled.Web.Sample/Startup.cs (hunk 1 above)
Comment (kirkone) [reply src/KK.DotNet.BackgroundTasks.Scheduled.Web.Sample/Startup.cs:88]: This is described [here](https://github.com/kirkone/KK.DotNet.BackgroundTasks.Scheduled#options)

File: src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs (hunk 2 above)
Comment (kirkone) [reply src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs]: implemented in fc33a7bb9de95152947d72badacc9cc9a33ace6e

File: src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs (hunk 3 above)
Comment (kirkone) [reply src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs:26]: implemented in 89f1f8a933dbf00fbcc5e1a8ba9f2653467ed73c

File: src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs (hunk 4 above)
Comment (kirkone) [reply src/KK.DotNet.BackgroundTasks.Scheduled/SchedulerExtensions.cs:54]: implemented in 89f1f8a933dbf00fbcc5e1a8ba9f2653467ed73c

**Review by paule96 — APPROVED** [review]
okay

**Pull request merged**